aws/Dockerfile

```diff
@@ -2,4 +2,8 @@ ARG REPOSITORY_BASE_PATH
 
 FROM ${REPOSITORY_BASE_PATH}:latest
 
-RUN pip install boto3 botocore
+USER root
+
+RUN apk add py3-boto3
```

Comment:
I think you should install boto3 and other deps for gcp using pip with `--break-system-packages`. Just like we do in the base Dockerfile for installing ansible.
We should avoid mixing apk and pip install for python deps.

Reply:
ah right, didn't even notice we use the break package there. yeah let's do that.

Reply:
This image is poorly designed TBH, we should rework it a bit. It was already a pain last time when we added version pinning.

Reply:
why is it poorly designed? how can we improve it?

Reply:
- We don't have versioning of ansible versions exposed as tags.
- The base image is exported and used as it. We should instead keep base to be a common building block, removes the `USER spacelift` from it. And have generic / was / gcp builds targets
- We do not control the version of python installed 😨 
- We inherit the base image from `public.ecr.aws/spacelift/runner-terraform`, which does not makes sense. We do not need any terraform thing in there.
- We can run [`dive`](https://github.com/wagoodman/dive) on it, but I think the size of the image contains tons of useless tools, which make it more easy to perform a supply chain attack on.
- We should add more test than just asserting on ansible binaries existence.

Reply:
🤔